plugins/ipam/dhcp/main.go

```diff
@@ -30,7 +32,18 @@ const socketPath = "/run/cni/dhcp.sock"
 
 func main() {
 	if len(os.Args) > 1 && os.Args[1] == "daemon" {
-		runDaemon()
+		var pidfilePath string
+		daemonFlags := flag.NewFlagSet("daemon", flag.ExitOnError)
+		daemonFlags.StringVar(&pidfilePath, "pidfile", "", "optional path to write daemon PID to")
+		if err := daemonFlags.Parse(os.Args[2:]); err != nil {
```

Comment:
With `flag.ExitOnError` provided above, this conditional will never trigger, since the `flag` library will call `os.Exit(2)` for you ([see here](https://golang.org/pkg/flag/#ErrorHandling)).  So it's ok to just drop the error in the application code.

Alternately, if you want to customize the error handling, pass `flag.ContinueOnError` into `NewFlagSet()`